src/Microsoft.PowerShell.Commands.Utility/commands/utility/join-string.cs

```diff
@@ -106,40 +106,43 @@ protected override void BeginProcessing()
         /// <inheritdoc />
         protected override void ProcessRecord()
         {
-            if (InputObject != null && InputObject != AutomationNull.Value)
+            foreach (PSObject inputObject in InputObject)
```

Comment:
What if inputObject is null?

Reply:
It is still there - now at line 111.

Reply:
In the Line 109 throw will be rised if InputObject is null.